itou/rdv_insertion/api.py

```diff
@@ -0,0 +1,41 @@
+import json
+
+import httpx
+from django.conf import settings
+from django.core.cache import cache
+from django.core.exceptions import ImproperlyConfigured
+
+
+def get_api_credentials(refresh=False):
+    """
+    RDV-I and RDV-S APIs share the same userbase and credentials.
+    Their tokens are issued for a 24h period and each renewal invalidates anterior tokens.
+    The credentials are cached and automatically refreshed using this function.
+    """
+    if settings.RDV_SOLIDARITES_API_BASE_URL and settings.RDV_SOLIDARITES_EMAIL and settings.RDV_SOLIDARITES_PASSWORD:
+        with cache.lock("rdv-solidarites-credentials-lock", blocking_timeout=10):
+            if refresh or not (api_credentials := cache.get("rdv-solidarites-credentials")):
```

Comment:
```suggestion
            if refresh or not (api_credentials_as_json := cache.get("rdv-solidarites-credentials")):
```
Parce que tu utilises actuellement le même nom pour le json et son contenu

Reply:
En fait la sérialisation JSON n'est pas nécessaire, le cache pickle par défaut.
Vu que ce n'est pas de l'user input et que ça reste assez simple, pickle fait le job :)